src/HX711Multi.py

```diff
@@ -0,0 +1,238 @@
+            outliers-from-a-numpy-array-in-python
+        """
+
+        samples = np.array(samples).T
+        means = np.mean(samples, axis=1).reshape(-1,1)
+        std_devs = np.std(samples, axis=1).reshape(-1,1)
+        dists_from_means = abs(samples - means)
+
+        # True/false mask to extract non outlier values
+        no_outliers = dists_from_means <= (max_std_dev_from_mean * std_devs)
+        clean_data = [row[no_outliers[i]] for i,row in enumerate(samples)]
+
+        clean_data_length = 0
+        for data_list in clean_data:
+            clean_data_length += len(data_list)
+
+        if len(clean_data) < 0.8*len(samples):
+            if self.__debug_enabled:
+                print("Failed to tare. Excessive deviations measured.")
+            return false
+            
+        # Calculate offsets from clean data
+        avg_samples = [np.average(row) for row in clean_data]
+        self.__offsets = avg_samples
+
+        if self.__debug_enabled:
+            print("Tared, offsets:")
+            print(self.__offsets)
+
+        return True
+
+    def read(self):
+        raw_readings = self.readRaw()
+        
+        if raw_readings == False:
+           return False 
+        else:
+            return np.array(raw_readings) - np.array(self.__offsets)
+
+    def readRaw(self):
+
+        if self.__debug_enabled and not self.isReady():
+            print("HX711s are not ready! Ensure isReady() returns True",
+                  "before calling read() or readRaw()")
+
+        # Read value from every HX711. Each cycle of PD_SCK shifts one bit
+        # out in twos complement.
+        readings = [0]*self.__num_of_HX711s
+
+        for bits in range(24):
+
+            GPIO.output(self.__PD_SCK_pin, GPIO.HIGH)
+            GPIO.output(self.__PD_SCK_pin, GPIO.LOW)
+
+            for i, data_pin in enumerate(self.__DOUT_pins):
+                readings[i] = readings[i] << 1
+                readings[i] |= GPIO.input(data_pin)
+
+        # Gain for the next reading is set by cycling the PD_SCK pin
+        # __gain number of times.
+        GPIO.output(self.__PD_SCK_pin, GPIO.LOW)
+
+        for _ in range(self.__gain):
+            GPIO.output(self.__PD_SCK_pin, GPIO.HIGH)
+            GPIO.output(self.__PD_SCK_pin, GPIO.LOW)
+            
+        # Calculate int from 2's complement as this is the form the HX711
+        # sends its data according to the datasheet. Readings are 24bit.
+        for reading in readings:
+            if (reading & (1 << (24 - 1))) != 0:
+                reading = reading - (1 << 24)
+
+        return readings
+
+    def __powerUp(self):
+        GPIO.output(self.__PD_SCK_pin, GPIO.LOW)
+
+    def __setGain(self, gain):
+
+        if gain == 128:
+            self.__gain = 1
+        elif gain == 64:
+            self.__gain = 3
+        elif self.__debug_enabled:
+            print("Invalid gain used. Pass values of 128 or 64 only!")
```

Comment:
```suggestion
            print("Invalid gain used. Pass values of 128 or 64 only!") 

```
Leave a new line at the end of the file. People do it in my work so it must be a thing

Reply:
I can't find anything online about this lmao. Can you ask your coworkers what the point of doing this is?

Reply:
Oh whoops I thought you wanted a space after the print statement haha. Blank line is probably for readability I guess? I'll add it in my next commit.